snuba/datasets/storages/events.py

```diff
@@ -345,16 +346,24 @@ def get_promoted_tags() -> Mapping[str, Sequence[str]]:
                 "tags": ChainMap(
                     {col.flattened: col.flattened for col in promoted_tag_columns},
                     {
-                        col.flattened.replace("_", "."): col.flattened
+                        col.flattened.replace("_", ".", 1): col.flattened
                         for col in promoted_context_tag_columns
                     },
                 ),
                 "contexts": {
-                    col.flattened.replace("_", "."): col.flattened
+                    col.flattened.replace("_", ".", 1): col.flattened
```

Comment:
This was a mistake. It forgot to carry over the `1` to replace only the first `.` inn the name  (see https://github.com/getsentry/snuba/pull/1019/files#diff-4ee24fa3be56815e9609657709e8135cR306) otherwise `device.model_id` becomes `device.model.id` instead of leaving as it is.

Reply:
Ouch. Why do we even want to translate the column names in this way? Should we have a utility function for this? There might be other places where this happens (like line 302) and I'm not sure if we're doing the right translation there either.

Reply:
302 is fine. I tested that when I saw this issue. But just because there is no column containing an underscore in promoted_context_tag_columns.
Why was this translation done this way before I am not sure. I will add the helper function.
Fortunately this is not ported to the errors storage so this will eventually be dead code.

Reply:
utility function added